miden-lib/asm/miden/kernel_invocation.masm

```diff
@@ -0,0 +1,375 @@
+# KERNEL PROCEDURE OFFSETS
+# =================================================================================================
+
+# OFFSET CONSTANTS
+# -------------------------------------------------------------------------------------------------
+
+# The offsets at which hash of the corresponding procedure is stored.
+# const.EXEC_KERNEL_PROC_OFFSET=0
+const.ACCOUNT_VAULT_GET_BALANCE_OFFSET=1
+const.GET_ACCOUNT_MAP_ITEM_OFFSET=2
+const.GET_ACCOUNT_ITEM_OFFSET=3
+const.GET_CURRENT_ACCOUNT_HASH_OFFSET=4
+const.GET_ACCOUNT_NONCE_OFFSET=5
+const.SET_ACCOUNT_ITEM_OFFSET=6
+const.GET_INITIAL_ACCOUNT_HASH_OFFSET=7
+const.MINT_ASSET_OFFSET=8
+const.GET_ACCOUNT_ID_OFFSET=9
+const.ACCOUNT_VAULT_HAS_NON_FUNGIBLE_ASSET_OFFSET=10
+const.BURN_ASSET_OFFSET=11
+const.INCR_ACCOUNT_NONCE_OFFSET=12
+const.CREATE_NOTE_OFFSET=13
+const.GET_FUNGIBLE_FAUCET_TOTAL_ISSUANCE_OFFSET=14
+const.GET_BLOCK_HASH_OFFSET=15
+const.GET_BLOCK_NUMBER_OFFSET=16
+const.ACCOUNT_VAULT_ADD_ASSET_OFFSET=17
+const.GET_NOTE_SERIAL_NUMBER_OFFSET=18
+const.GET_ACCOUNT_VAULT_COMMITMENT_OFFSET=19
+const.ACCOUNT_VAULT_REMOVE_ASSET_OFFSET=20
+const.SET_ACCOUNT_CODE_OFFSET=21
+const.GET_NOTE_ASSETS_INFO_OFFSET=22
+const.GET_NOTE_SENDER_OFFSET=23
+const.SET_ACCOUNT_MAP_ITEM_OFFSET=24
+const.GET_NOTE_INPUTS_HASH_OFFSET=25
+const.ADD_ASSET_TO_NOTE_OFFSET=26
+const.GET_OUTPUT_NOTES_HASH_OFFSET=27
+const.GET_INPUT_NOTES_COMMITMENT_OFFSET=28
```

Comment:
Probably not for this PR, but we should come up with a logical way to group these procedures. We should also figure out how to keep these constants consistent with the `TransactionKernel::PROCEDURES` constant I mentioned in one of my previous comments.